testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/java/org/keycloak/testsuite/services/clientpolicy/executor/TestDPoPSimulateExecutor.java

```diff
@@ -0,0 +1,102 @@
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutorProvider;
+
+/**
+ * @author <a href="mailto:takashi.norimatsu.ws@hitachi.com">Takashi Norimatsu</a>
+ */
+public class TestDPoPSimulateExecutor implements ClientPolicyExecutorProvider<ClientPolicyExecutorConfigurationRepresentation> {
+
+    public static final String DUMMY_JKT = "0ZcOCORZNYy-DWpqq30jZyJGHTN0d2HglBV3uiguA4I";
+
+    private static final Logger logger = Logger.getLogger(TestDPoPSimulateExecutor.class);
+
+    protected final KeycloakSession session;
+
+    private PublicKey DPoPPublicKey;
+
+    public TestDPoPSimulateExecutor(KeycloakSession session) {
+        this.session = session;
+    }
+
+    @Override
+    public String getProviderId() {
+        return TestDPoPSimulateExecutorFactory.PROVIDER_ID;
+    }
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        ClientPolicyEvent event = context.getEvent();
+        logger.tracev("Client Policy Trigger Event = {0}",  event);
+        switch (event) {
+            case TOKEN_RESPONSE:
+                verifyDPoPProof(context);
+                bindTokensWithDPoPPublicKey((TokenResponseContext)context);
+                break;
+            case TOKEN_REFRESH_RESPONSE:
+                verifyDPoPProof(context);
+                verifyBindingTokensWithDPoPPublicKey((TokenResponseContext)context);
+                bindTokensWithDPoPPublicKey((TokenResponseContext)context);
+                break;
+            default :
+                return;
+        }
+    }
+
+    private void verifyDPoPProof(ClientPolicyContext context) throws ClientPolicyException {
+        // retrieve a DPoP Proof JWT from the "DPoP" HTTP header
+        //String DPoPProofJwtString = session.getContext().getRequestHeaders().getHeaderString("DPoP");
+
+        // retrieve a DPoP public key from the DPoP Proof JWT
+
+        // verify a signature of DPoP Proof JWT with the DPoP public key
+        // if it fails, raise ClientPolicyException
+        //throw new ClientPolicyException("invalid_dpop_proof", "verifying a signature of DPoP Proof JWT with the DPoP public key failed.");
+    }
+
+    private void bindTokensWithDPoPPublicKey(TokenResponseContext context) {
+        // calculate "jkt" against the DPoP public key
+        String jkt = DUMMY_JKT;
+        // set cnf.jkt to an access token and a refresh token
+        ProtocolMapperModel mappingModel = HardcodedClaim.create("hard", "cnf.jkt", jkt, "String", true, true);
+        OIDCAttributeMapperHelper.mapClaim(context.getAccessTokenResponseBuilder().getAccessToken(), mappingModel, jkt);
+        OIDCAttributeMapperHelper.mapClaim(context.getAccessTokenResponseBuilder().getRefreshToken(), mappingModel, jkt);
+    }
+
+    private void verifyBindingTokensWithDPoPPublicKey(TokenResponseContext context) {
+        // retrieve a DPoP public key from the DPoP Proof JWT
```

Comment:
Same comment as above

Reply:
As you suggested, I think this test should be done in the follow-up issue of the DPoP initial PR. Therefore, I will remove this test from the PR.